### ISSUE TITLE
0.3.2 breaks crossbuild

### ISSUE THREAD

**Issue by Gottox** [hroptatyr/dateutils#33]
Hello!

dateutils-0.3.2 breaks crossbuild.

configure output when building for armv7 on x86_64:

```
configure: WARNING: unrecognized options: --with-libtool-sysroot
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for arm-linux-gnueabihf7-strip... arm-linux-gnueabihf7-strip
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for arm-linux-gnueabihf7-icc... arm-linux-gnueabihf7-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether arm-linux-gnueabihf7-gcc accepts -g... yes
checking for arm-linux-gnueabihf7-gcc option to accept ISO C89... none needed
checking for style of include used by make... GNU
checking dependency style of arm-linux-gnueabihf7-gcc... gcc3
checking build system type... x86_64-unknown-linux-gnu
checking host system type... arm-unknown-linux-gnueabihf7
checking how to run the C preprocessor... arm-linux-gnueabihf7-cpp
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking whether C compiler accepts -std="gnu11"... yes
checking whether C compiler can understand anonymous structs and unions... yes
checking for preferred CC std... -std=gnu11
checking whether C compiler can initialise anonymous structs and unions... yes
checking whether C compiler accepts -Wmissing-field-initializers... yes
checking whether C compiler can initialise structs and unions in a sloppy way... no
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking whether byte ordering is bigendian... (cached) no
checking for user provided CFLAGS/EXTRA_CFLAGS...  -O2 -pipe -fstack-protector-strong -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard   
checking whether C linker accepts -XCClinker -foo... no
checking whether C linker accepts -Xlinker -foo... no
checking whether C compiler accepts -g... yes
checking whether C compiler accepts -g3... yes
checking for dbx... no
checking for gdb... no
checking for sdb... no
checking whether C compiler accepts -ftime-report... no
checking whether C compiler accepts -fmem-report... no
checking whether C compiler accepts -fvar-tracking... no
checking whether C compiler accepts -save-temps... yes
checking for preferred debugging flags... 
checking whether C compiler accepts -Wall... yes
checking whether C compiler accepts -qinfo... no
checking whether C compiler accepts -Wextra... yes
checking whether C compiler accepts -Wsign-compare... yes
checking whether C compiler accepts -Wno-char-subscripts... yes
checking whether C compiler accepts -Wundef... yes
checking whether C compiler accepts -Wpacked... yes
checking whether C compiler accepts -Wpointer-arith... yes
checking whether C compiler accepts -Wshadow... yes
checking whether C compiler accepts -Wmissing-declarations... yes
checking whether C compiler accepts -Wmissing-prototypes... yes
checking whether C compiler accepts -Wbad-function-cast... yes
checking whether C compiler accepts -Wcast-qual... yes
checking whether C compiler accepts -Wcast-align... yes
checking whether C compiler accepts -Wno-switch... yes
checking whether C compiler accepts -Wunused-function... yes
checking whether C compiler accepts -Wunused-variable... yes
checking whether C compiler accepts -Wunused-parameter... yes
checking whether C compiler accepts -Wunused-value... yes
checking whether C compiler accepts -Wunused... yes
checking whether C compiler accepts -Wmaybe-uninitialized... yes
checking whether C compiler accepts -Wnopragma... no
checking whether C compiler accepts -fdiagnostics-show-option... yes
checking whether C compiler accepts -Wunknown-pragmas... yes
checking whether C compiler accepts -Wuninitialized... yes
checking whether C compiler accepts -Wreorder... no
checking whether C compiler accepts -Wdeprecated... yes
checking whether C compiler accepts -Wno-parentheses... yes
checking whether C compiler accepts -Wcheck... no
checking whether C compiler accepts -Wstrict-aliasing... yes
checking whether C compiler accepts -w3... no
checking whether C compiler accepts -diag-disable 10237... no
checking whether C compiler accepts -wd 10237... no
checking whether C compiler accepts -diag-disable 2102... no
checking whether C compiler accepts -wd 2102... no
checking whether C compiler accepts -debug inline-debug-info... no
checking whether C compiler accepts -diag-enable remark,vec,par... no
checking whether C compiler accepts -Wunsuffixed-float-constants... yes
checking for preferred warning flags... -Wall -Wextra -Wsign-compare -Wno-char-subscripts -Wundef -Wpacked -Wpointer-arith -Wshadow -Wmissing-declarations -Wmissing-prototypes -Wbad-function-cast -Wcast-qual -Wcast-align -Wno-switch -Wunused-function -Wunused-variable -Wunused-parameter -Wunused-value -Wunused -Wmaybe-uninitialized -fdiagnostics-show-option -Wunknown-pragmas -Wuninitialized -Wdeprecated -Wno-parentheses -Wstrict-aliasing -Wunsuffixed-float-constants
checking whether C compiler accepts -ipo... no
checking whether C compiler accepts -no-prec-div... no
checking whether C compiler accepts -nopie... no
checking whether C compiler accepts -static-intel... no
checking whether C compiler accepts -static-libgcc... yes
checking whether C compiler accepts -intel-extensions... no
checking for preferred CFLAGS...  -O2 -pipe -fstack-protector-strong -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard      -Wall -Wextra -Wsign-compare -Wno-char-subscripts -Wundef -Wpacked -Wpointer-arith -Wshadow -Wmissing-declarations -Wmissing-prototypes -Wbad-function-cast -Wcast-qual -Wcast-align -Wno-switch -Wunused-function -Wunused-variable -Wunused-parameter -Wunused-value -Wunused -Wmaybe-uninitialized -fdiagnostics-show-option -Wunknown-pragmas -Wuninitialized -Wdeprecated -Wno-parentheses -Wstrict-aliasing -Wunsuffixed-float-constants  -static-libgcc 
configure:
If you wish to APPEND your own flags you want to stop here and rerun the
configure script like so:
  configure EXTRA_CFLAGS=<to-be-added-flags>

If you wish to OVERRIDE these flags you want to stop here too and rerun
the configure script like this:
  configure CFLAGS=<the-definitive-flags-I-want>

You can always override the determined CFLAGS, partially or totally,
using
  make -C <directory> CFLAGS=<your-own-flags> target
or
  make CFLAGS=<your-own-flags> target
respectively
		
checking for arm-linux-gnueabihf7-gcc -std=gnu11 option to accept ISO C99... none needed
checking for stipulated version files... configure: error: in `/builddir/dateutils-0.3.2':
configure: error: cannot run test program while cross compiling
See `config.log' for more details
=> ERROR: dateutils-0.3.2_1: failed to run do_configure() at line 7.
```

workaround:

```
sed -i "/^## check\/generate version files/,/^## check for byteorder utils/d" configure
cat > version.mk <<EOF
VERSION = 0.3.2
PACKAGE_VERSION = \$(VERSION)
PACKAGE_STRING = \$(PACKAGE) \$(VERSION)
EOF
```

**Comment by hroptatyr** [comment]
Can you send the config.log output?  I'm not too familiar with cross builds.

**Comment by hroptatyr** [comment]
Sorted it with a big hammer, should be fixed in ee376ee.

**Comment by Gottox** [comment]
This fixes cross compilation.

https://github.com/voidlinux/void-packages/commit/dd691598e1ed9d1a3efcbe99be54549bffeb8145 
http://build.voidlinux.eu/builders/x86_64_builder/builds/10490/steps/shell_3/logs/stdio

thanks :)
